### PR TITLE
Add support for POCO inserts

### PR DESCRIPTION
**Pull request by alex-clickhouse**
Right now the binary insert requires users to provide `object[][]`. This PR lets them use POCOs instead.

 * New `InsertBinaryAsync<T>()` method for inserting.
 * Two new property attributes: `[ClickHouseColumn(Name, Type)]` for setting column name/type, `[ClickHouseNotMapped]` for exclusion. If all properties are covered, we skip the schema probe.
 * Type registration required via `RegisterBinaryInstertType<T>().` This saves the serialization info and compiles property accessors into `Func<T, object>` delegates to avoid reflection when we're reading the properties in the serialization loop.
 * Internally, extracted shared insert prep functionality into `PrepareInsertAsync()`.
 * Added new example..

New benchmark added to compare object[][] and POCO serialization. Current results:

| Method      | Count  | Mean      | Error    | StdDev    | Ratio | RatioSD | Gen0       | Gen1      | Gen2      | Allocated | Alloc Ratio |
|------------ |------- |----------:|---------:|----------:|------:|--------:|-----------:|----------:|----------:|----------:|------------:|
| ObjectArray | 100000 |  47.17 ms | 3.338 ms |  7.185 ms |  1.02 |    0.22 |  2000.0000 | 1000.0000 |         - |  13.76 MB |        1.00 |
| Poco        | 100000 |  51.21 ms | 5.293 ms | 11.839 ms |  1.11 |    0.31 |  2000.0000 | 1000.0000 |         - |     13 MB |        0.94 |
|             |        |           |          |           |       |         |            |           |           |           |             |
| ObjectArray | 500000 | 198.00 ms | 3.673 ms |  7.828 ms |  1.00 |    0.06 | 12000.0000 | 7000.0000 | 2000.0000 |  66.72 MB |        1.00 |
| Poco        | 500000 | 177.00 ms | 2.992 ms |  6.178 ms |  0.90 |    0.05 | 12000.0000 | 7000.0000 | 2000.0000 |  62.91 MB |        0.94 |

Related docs PR: https://github.com/ClickHouse/clickhouse-docs/pull/5919

### CODE REVIEW

File: ClickHouse.Driver.Tests/Copy/BinaryInsertTypeRegistryTests.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using ClickHouse.Driver.ADO;
+using NUnit.Framework;
+
+namespace ClickHouse.Driver.Tests.Copy;
+
+[TestFixture]
+public class BinaryInsertTypeRegistryTests
```
Comment (mzitnik) [inline ClickHouse.Driver.Tests/Copy/BinaryInsertTypeRegistryTests.cs:10]: I think we should add a few more edge cases 

- two columns with the same name
- column with reserved name

File: ClickHouse.Driver/Copy/Serializer/PocoBatchSerializer.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    /// <summary>
+    /// Serializes a batch of POCO rows into the target stream using GZip compression.
+    /// </summary>
+    /// <typeparam name="T">The POCO type.</typeparam>
+    /// <param name="batch">The batch of rows, query text, and resolved column types.</param>
+    /// <param name="getters">Compiled property accessors, ordered to match the batch's column types.</param>
+    /// <param name="stream">The output stream (typically a recyclable memory stream).</param>
+    public void Serialize<T>(PocoBatch<T> batch, Func<T, object>[] getters, Stream stream)
+    {
+        using var gzipStream = new BufferedStream(new GZipStream(stream, CompressionLevel.Fastest, true), 256 * 1024);
```
Comment (mzitnik) [inline ClickHouse.Driver/Copy/Serializer/PocoBatchSerializer.cs:41]: i tihnk we should have some of the params of `GZipStream` configerable

Comment (alex-clickhouse) [reply ClickHouse.Driver/Copy/Serializer/PocoBatchSerializer.cs:41]: Sounds good but let's leave that for another PR. I added a story for it.

File: ClickHouse.Driver.Tests/Copy/BinaryInsertTypeRegistryTests.cs (hunk 1 above)
Comment (alex-clickhouse) [reply ClickHouse.Driver.Tests/Copy/BinaryInsertTypeRegistryTests.cs:10]: >two columns with the same name

Already covered by `PocoWithDuplicateColumnNames`.

**Review by mzitnik — APPROVED** [review]
